### ISSUE TITLE
[Linux] add support PyFilesystem2

### ISSUE THREAD

**Issue by Tsvetk** [giampaolo/psutil#2132]
## Summary

* OS: { Linux }
* Type: {  }

## Description

{{{
Add support PyFilesystem2 with fs.sshfs

https://github.com/PyFilesystem/pyfilesystem2
https://pypi.org/project/fs.sshfs/

I  made small modification  :
def cpu_stats(fs):
    """Return various CPU stats as a named tuple."""
    with **fs.openbin('/stat')** as f:
        ctx_switches = None
        interrupts = None

Write script  for test 

my_fs = fs.open_fs("ssh://root:secret_pass@host/proc")

print(cpu_stats(my_fs))
scpustats(ctx_switches=284272841, interrupts=617737931, soft_interrupts=40807609, syscalls=0)
print(cpu_stats(my_fs))
scpustats(ctx_switches=284274774, interrupts=617742066, soft_interrupts=40808011, syscalls=0)
print(cpu_stats(my_fs))
scpustats(ctx_switches=284276892, interrupts=617746567, soft_interrupts=40808432, syscalls=0)

All work

}}}

**Comment by giampaolo** [comment]
Nope. This would add a third party dep (a first in psutil), plus it cannot work consistently across all APIs, since some APIs rely on `/proc` FS, others invoke kernel syscalls in C. As such psutil cannot make a promise that **all** APIs will return info about the remote system.

With that said, psutil already gives the possibility to do what you ask for: you can mount a network dir pointing to the remote /proc fs and do:

```python
import psutil
psutil.PROCFS_PATH = "/remote/network/drive/procfs"
```
